### PR TITLE
Make the other installation suppliers work for daemon toolchain detection too

### PR DESCRIPTION
**Pull request by big-guy**
### Reviewing cheatsheet

Before merging the PR, comments starting with 
- ❌ ❓**must** be fixed
- 🤔 💅 **should** be fixed
- 💭 **may** be fixed
- 🎉 celebrate happy things

### CODE REVIEW

**Comment by big-guy** [comment]
@bot-gradle test this

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc [hunk 1]
```diff
@@ -468,3 +468,21 @@ include::sample[dir="snippets/tutorial/environmentVariables/groovy",files="init.
...
 ====
+
+[[sec:daemon_jvm_criteria]]
+== Controlling JVM used by the build
+
+NOTE: Daemon JVM discovery and Daemon JVM criteria are <<feature_lifecycle.adoc#sec:incubating_state,incubating>> features and are subject to change in a future release.
+
+By default, Gradle will try to use the same JVM installation to run the build as the client VM that starts the build.
+
+A different JVM installation can be specified for the build by using the `org.gradle.java.home` property or programmatically through the Tooling API.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc:479]:
```suggestion
A different JVM installation can be specified for the build using the `org.gradle.java.home` property or programmatically through the Tooling API.
```

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc [hunk 2]
```diff
@@ -468,3 +468,21 @@ include::sample[dir="snippets/tutorial/environmentVariables/groovy",files="init.
...
+
+Daemon JVM criteria is a new feature to express the build's JVM requirements.
+In a similar way to the `wrapper` task capturing the current Gradle version, the built-in task `updateDaemonJvm` records the JVM requirements for the build based on the current JVM installation.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc:482]:
```suggestion
Similar to the `wrapper` task, which captures the current Gradle version, the built-in `updateDaemonJvm` task records the JVM requirements for the build based on the current JVM installation.
```

File: platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc [hunk 3]
```diff
@@ -468,3 +468,21 @@ include::sample[dir="snippets/tutorial/environmentVariables/groovy",files="init.
...
+Gradle currently only considers the major JVM version when checking compatibility.
+
+On subsequent builds, Gradle will use the JVM requirements to locate a compatible JVM installation using the same underlying mechanisms as <<toolchains.adoc, Java Toolchains>>.
```
Comment (lkasso) [inline platforms/documentation/docs/src/docs/userguide/authoring-builds/other/build_environment.adoc:485]:
```suggestion
Gradle will use the JVM requirements on subsequent builds to locate a compatible JVM installation using the same underlying mechanisms as <<toolchains.adoc, Java Toolchains>>.
```

**Review by lkasso — APPROVED** [review]
Minor language changes in docs.

**Comment by big-guy** [comment]
@bot-gradle merge

**Pull request merged**